calamus/fields.py

```diff
@@ -86,40 +92,35 @@ def __str__(self):
 class _JsonLDField(fields.Field):
     """Internal class that enables marshmallow fields to be serialized with a JsonLD field name."""
 
-    def __init__(self, field_name, *args, **kwargs):
+    def __init__(self, field_name=None, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.field_name = field_name
 
         self.reverse = kwargs.get("reverse", False)
+        self.init_name = kwargs.get("init_name", None)
 
     @property
     def data_key(self):
         """Return the (expanded) JsonLD field name."""
+        if self.field_name is None:
+            raise ValueError("field_name was not set for {}".format(self))
```

Comment:
I think we should add a default `_-repr__` and possibly `__str__` to `_JsonLDField`, otherwise printing a field will print something like `<fields.String(default=<marshmallow.missing>, attribute=None, validate=None, required=False, load_only=False, dump_only=False, missing=<marshmallow.missing>, allow_none=False, error_messages={'required': 'Missing data for required field.', 'null': 'Field may not be null.', 'validator_failed': 'Invalid value.', 'invalid': 'Not a valid string.', 'invalid_utf8': 'Not a valid utf-8 string.'})>` which only shows marshmallow fields.

Doesn't help in this instance since there's no field_name anyways, but we could change the message here to `field_name was not set for {} in schema {}` using self.parent (not sure how to handle `fields.List`), then at least you know in what schema you missed setting this

Reply:
`self.name` is the name of this field in the schema and `self.root` is the schema (works for `fields.List` too). I've changed this to:
```
"field_name was not set for {} in schema {}".format(self.name, self.root.__class__)
```